src/Cards.cpp

```diff
@@ -0,0 +1,141 @@
+//
+// Created by claudia on 9/12/2019.
+//
+
+#include <random>
+#include <vector>
+#include <algorithm>
+#include "../include/Cards.h"
+
+namespace globalHandVariables {
+    int tradedSetsCount = 1;
+}
+
+/**
+ * Deck Constructor
+ * @param amountCountries
+ */
+Deck::Deck(int amountCountries) {
+    deckSize = new int(amountCountries); //store value on the heap to avoid losing it
+    deckPointer = new std::vector<CardType>;
+    discardPointer = new std::vector<CardType>;
+}
+
+/**
+ * Hand Constructor
+ */
+Hand::Hand() {
+    handPointer = new std::vector<CardType>;
+}
+
+/**
+ * Populates a deck of size equal to the number of countries
+ * @return
+ */
+void Deck::createDeck() {
+    std::vector<CardType> deck = Deck::populateDeck();
+    std::shuffle(deck.begin(), deck.end(), std::mt19937(std::random_device()()));
+    deckPointer = new std::vector<CardType>(deck); //store value on the heap to avoid losing it
+}
+
+/**
+ * Add as many cards as there are counties to the game deck. Equal parts infantry, cavalry, and artillery
+ * @return
+ */
+std::vector<CardType> Deck::populateDeck() {
+
+    std::vector<CardType> deck = {};
+
+    for (int i = 0; i < Deck::getNumberOfCards() / 3; i++) {
+        deck.emplace_back(CardType::INFANTRY);
+        deck.emplace_back(CardType::ARTILLERY);
+        deck.emplace_back(CardType::CAVALRY);
+    }
+
+    Deck::setNumberOfCards(deck.size());
+
+    return deck;
+}
+
+/**
+ * User draws a random card from the deck and it is removed from the deck
+ */
+void Deck::draw(Hand &userHand) {
+    CardType drawnCard = deckPointer->back();
+    deckPointer->pop_back();
+    Deck::setNumberOfCards(deckPointer->size());
+    userHand.getHand()->push_back(drawnCard);
+}
+
+/**
+ * exchange three cards of kind of three different cards to gain army members
+ * @param deck
+ * @return
+ */
+int Hand::exchange(Hand hand, Deck deck, const std::vector<CardType> & givenCards) {
+
+    int armiesToExchange = Hand::armiesReceived();
+
+    if (givenCards.size() != 3) {
+        return -1;
+    }
+
+    //Ensure that all three given cards are the same
+    if (givenCards.at(0) == givenCards.at(1) && givenCards.at(1) == givenCards.at(2)) {
+        deck.discard(hand, givenCards);
+        globalHandVariables::tradedSetsCount++;
+        return armiesToExchange;
+    }
+        //Ensure that all three cards are different
+    else if (givenCards.at(0) != givenCards.at(1) &&
+             givenCards.at(0) != givenCards.at(2) &&
+             givenCards.at(1) != givenCards.at(2)) {
+        deck.discard(hand, givenCards);
+        globalHandVariables::tradedSetsCount++;
+        return armiesToExchange;
+    }
+    return -1;
+}
+
+/**
+ * Determine the number of armies received through card exchange
+ * @return
+ */
+int Hand::armiesReceived() {
+
+    if (globalHandVariables::tradedSetsCount < 6) {
+        return (globalHandVariables::tradedSetsCount * 2 + 2);
```

Comment:
How come the parenthesis around the return? 
```suggestion
        return globalHandVariables::tradedSetsCount * 2 + 2;
```

Reply:
I just find it cleaner. Can remove it

Reply:
It's not a big deal really if you'd like to keep it 🤷‍♂